content/en/blog/falco-plugin-k8saudit-eks.md

```diff
@@ -51,68 +56,314 @@ With:
+14:09:24.046826000: Notice Attach/Exec to pod (user=kubernetes-admin pod=kubecon resource=pods ns=default action=exec command=sh)
 ```
 
-The referencies to `eks` proves it works!!!
+### Kubernetes installation
+
+Let's see now how to install the `k8saudit-eks` plugin with `falcoctl` tool. We'll use the official Helm chart with this `values.yaml` file:
+```yaml
+serviceAccount:
+  create: true
+  annotations:
+    - eks.amazonaws.com/role-arn: {ROLE_ARN}
+
+falco:
+  rules_file:
+    - /etc/falco/k8s_audit_rules.yaml
+    - /etc/falco/rules.d
+  plugins:
+    - name: k8saudit-eks
+      library_path: libk8saudit-eks.so
+      init_config:
+        shift: 10
+        polling_interval: 10
+        use_async: false
+        buffer_size: 500
+      open_params: "{CLUSTER_NAME}"
+    - name: json
+      library_path: libjson.so
+      init_config: ""
+  load_plugins: [k8saudit-eks, json]
+
+falcosidekick:
+  enabled: true
+
+driver:
+  enabled: false
+collectors:
+  enabled: false
+
+controller:
+  kind: deployment
+  deployment:
+    replicas: 1
+
+falcoctl:
+  indexes:
+  - name: falcosecurity
+    url: https://falcosecurity.github.io/falcoctl/index.yaml
+  artifact:
+    install:
+      enabled: true
+    follow:
+      enabled: true
+  config:
+    artifact:
+      allowedTypes:
+        - plugin
+        - rulesfile
+      install:
+        resolveDeps: false
+        refs: [k8saudit-rules:0, k8saudit-eks:0, json:0]
+      follow:
+        refs: [k8saudit-rules:0]
+```
+
+Let's explain this explain section by section:
+```yaml
+serviceAccount:
+  create: true
+  annotations:
+    - eks.amazonaws.com/role-arn: {ROLE_ARN}
+```
+With EKS, we can use an OIDC provider and annotations on the service accounts to allow the access to the AWS API. See [the official docs](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html). This section allows you to set the `ROLE_ARN` to use.
+
+```yaml
+falco:
+  rules_file:
+    - /etc/falco/k8s_audit_rules.yaml
+    - /etc/falco/rules.d
+  plugins:
+    - name: k8saudit-eks
+      library_path: libk8saudit-eks.so
+      init_config:
+        shift: 10
+        polling_interval: 10
+        use_async: false
+        buffer_size: 500
+      open_params: "{CLUSTER_NAME}"
+    - name: json
+      library_path: libjson.so
+      init_config: ""
+  load_plugins: [k8saudit-eks, json]
+```
+
+Like for the local installation, this section allows to set the usage of the plugins.
+
+```yaml
+driver:
+  enabled: false
+collectors:
+  enabled: false
+
+controller:
+  kind: deployment
+  deployment:
+    replicas: 1
+```
+
+{{% pageinfo color="warning" %}}
+Falco must be installed as a DaemonSet to have one instance per node in your cluster, but for the `k8saudit-eks` plugin, we MUST run only 1 pod to avoid to pull same logs several times and duplicate alert. This is why we deploy Falco as a 1 replica deployment. The syscalls and Kubernetes metadata collections are also disabled.
```

Comment:
```suggestion
Notice that we deploy Falco as a 1 replica deployment. For the `k8saudit-eks` plugin, we MUST run only 1 pod to avoid pulling the same logs several times and duplicating alerts. This deployment is different from the system-call one, in which Falco is installed as a DaemonSet to have one instance per node in your cluster. The system-call deployment is disabled along with the Kubernetes metadata collection.
```

Reply:
I am still not happy with the result. There is too much information here if you are not familiar with it. Would it make sense to break the explanation into two parts? Or do they only make sense together? Could I have a single configuration that enables the system-call deamonset deployment AND the 1 replicaset deployment?

Reply:
We can't deploy falco in the 2 methods with a single values.yaml. This is why I'm explaining. Let me find a better explanation.